src/main/java/games/strategy/util/Util.java

```diff
@@ -122,4 +130,19 @@ public static String getStringFromInputStream(InputStream in) {
     }
     return builder.toString();
   }
+
+  /**
+   * Creates a hash of the given String based on the SHA-512 algorithm.
+   * 
+   * @param input The input String to hash.
+   * @return A hashed hexadecimal String of the input.
+   */
+  public static String sha512(String input) {
```

Comment:
Does the server continue to only use the SHA-512 hashed value? I'm a bit confused exactly now where the decryption comes in. What I'm driving towards is to avoid the server from doing decryption. Since we do not have TLS, it makes sense to hash the password before sending, then IMO that value *becomes* effectively the password itself. I think we both agree the goal is so only the original password holder knows their password, not even devs.
 To be sure, in that model, the SHA-512 value is hashed again to get the final value we would store in DB. I suspect I may misunderstand somewhere, please let me know where I go wrong, if not, could you confirm the decryption the server is doing?